modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/WatchSync.kt

```diff
@@ -0,0 +1,135 @@
+package au.com.shiftyjelly.pocketcasts.account
+
+import android.annotation.SuppressLint
+import android.app.Activity
+import android.content.Context
+import au.com.shiftyjelly.pocketcasts.preferences.Settings
+import au.com.shiftyjelly.pocketcasts.utils.log.LogBuffer
+import com.google.android.gms.wearable.DataEvent
+import com.google.android.gms.wearable.DataEventBuffer
+import com.google.android.gms.wearable.DataItem
+import com.google.android.gms.wearable.DataMapItem
+import com.google.android.gms.wearable.PutDataMapRequest
+import com.google.android.gms.wearable.PutDataRequest
+import com.google.android.gms.wearable.Wearable
+import dagger.hilt.android.qualifiers.ApplicationContext
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.tasks.await
+import kotlinx.coroutines.withContext
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+@SuppressLint("VisibleForTests") // https://issuetracker.google.com/issues/239451111
+class WatchSync @Inject constructor(
+    @ApplicationContext context: Context,
+    private val settings: Settings,
+    private val accountAuth: AccountAuth,
+) {
+
+    companion object {
+        private const val authPath = "/auth"
+        private const val authKeyRefreshToken = "refreshToken"
+    }
+
+    private val dataClient = Wearable.getDataClient(context)
+
+    /**
+     * This should be called by the phone app to update the refresh token available to
+     * the watch app in the data layer.
+     */
+    suspend fun sendAuthToDataLayer(activity: Activity) {
+        withContext(Dispatchers.Default) {
+            try {
+                Timber.i("Updating refresh token in data layer")
+
+                val authData = let {
+                    val email = settings.getSyncEmail()
+                    val password = settings.getSyncPassword()
+                    if (email != null && password != null) {
+                        // FIXME nonononono this makes an api call _every_ time
+                        accountAuth.getTokensWithEmailAndPassword(email, password)
```

Comment:
Is the intention to stop storing the password in `AccountManager` altogether, and store the refresh token instead? It seems like that would be the most straightforward way to keep track of everything. Only if the user does not have a refresh token stored would you check if there's a password stored. The less state to keep track of, the better.

Let me know if I'm overstepping my bounds at all with this sort of feedback.

Reply:
> Is the intention to stop storing the password in AccountManager altogether, and store the refresh token instead?

Yes, I believe that will be a part of @geekygecko 's work on [Google Single-On work](https://github.com/Automattic/pocket-casts-android/pull/712). The api call on this line to retrieve the refresh token is just a placeholder until that work is done and we can retrieve the token locally.